validation/validation_test.go

```diff
@@ -127,7 +120,8 @@ func TestEmailDomainValidationCorrectlyValidatesAllowedDomains(t *testing.T) {
 		emailResponse := IsAllowedEmailDomain(email, allowedDomains)
 		So(emailResponse, ShouldBeFalse)
 
-		newAllowedDomains := append(allowedDomains, "@gmail.com")
+		newAllowedDomains := append([]string{}, allowedDomains...)
+		newAllowedDomains = append(newAllowedDomains, "@gmail.com")
```

Comment:
Theres a function in the slices package that might be useful.
```suggestion
		newAllowedDomains := append(slices.Clone(allowedDomains), "@gmail.com")
```